indexing-service/src/main/java/io/druid/indexing/overlord/TasksAndWorkers.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.indexing.overlord;
+
+import com.google.common.base.Predicate;
+import io.druid.indexing.common.task.Task;
+import io.druid.indexing.overlord.config.WorkerTaskRunnerConfig;
+import io.druid.indexing.worker.Worker;
+
+import java.util.Collection;
+
+public interface TasksAndWorkers
```

Comment:
I didn't want to make `TasksAndWorkersFilteredByIp` to implement `WorkerTaskRunner` because there are a lot of methods in `TaskRunner` which don't make sense for the interface, needed in Provisioners. And not clear how to implement them.

I agree that something is wrong here and some refactoring is needed, but it would affect a lot of unrelated code, so I don't want to do this until a PR is done into Druid upstream.